File: packages/go/luna/testutil/fixtures.go
package testutil

import (
	"encoding/json"
	"time"
)

// User fixtures
var MockUser = map[string]interface{}{
	"id":         "usr_123456789",
	"name":       "John Doe",
	"email":      "john@example.com",
	"avatar":     "https://example.com/avatar.jpg",
	"created_at": time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
	"updated_at": time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
}

var MockUsers = []map[string]interface{}{
	MockUser,
	{
		"id":         "usr_987654321",
		"name":       "Jane Smith",
		"email":      "jane@example.com",
		"avatar":     nil,
		"created_at": time.Date(2024, 1, 2, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
		"updated_at": time.Date(2024, 1, 2, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
	},
}

var MockUserCreate = map[string]string{
	"name":  "New User",
	"email": "newuser@example.com",
}

// Project fixtures
var MockProject = map[string]interface{}{
	"id":          "prj_123456789",
	"name":        "Test Project",
	"description": "A test project for unit tests",
	"owner_id":    "usr_123456789",
	"created_at":  time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
	"updated_at":  time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
}

var MockProjects = []map[string]interface{}{
	MockProject,
	{
		"id":          "prj_987654321",
		"name":        "Another Project",
		"description": "Another test project",
		"owner_id":    "usr_987654321",
		"created_at":  time.Date(2024, 1, 2, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
		"updated_at":  time.Date(2024, 1, 2, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
	},
}

// Storage fixtures
var MockBucket = map[string]interface{}{
	"id":         "bkt_123456789",
	"name":       "test-bucket",
	"public":     false,
	"created_at": time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
}

var MockBuckets = []map[string]interface{}{
	MockBucket,
	{
		"id":         "bkt_987654321",
		"name":       "public-bucket",
		"public":     true,
		"created_at": time.Date(2024, 1, 2, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
	},
}

var MockFile = map[string]interface{}{
	"id":         "file_123456789",
	"name":       "test-file.pdf",
	"bucket_id":  "bkt_123456789",
	"size":       1024,
	"mime_type":  "application/pdf",
	"created_at": time.Date(2024, 1, 1, 0, 0, 0, 0, time.UTC).Format(time.RFC3339),
}

// Helper functions
func MockListResponse(data interface{}, hasMore bool, nextCursor string) map[string]interface{} {
	response := map[string]interface{}{
		"data":     data,
		"has_more": hasMore,
	}
	if nextCursor != "" {
		response["next_cursor"] = nextCursor
	}
	return response
}

// Error response fixtures
var MockErrorNotFound = map[string]interface{}{
	"error": map[string]interface{}{
		"message": "Resource not found",
		"code":    "NOT_FOUND",
		"status":  404,
	},
}

var MockErrorValidation = map[string]interface{}{
	"error": map[string]interface{}{
		"message": "Validation failed",
		"code":    "VALIDATION_ERROR",
		"status":  400,
		"details": []map[string]string{
			{"field": "email", "message": "Invalid email format"},
		},
	},
}

var MockErrorRateLimit = map[string]interface{}{
	"error": map[string]interface{}{
		"message":     "Rate limit exceeded",
		"code":        "RATE_LIMIT_EXCEEDED",
		"status":      429,
		"retry_after": 60,
	},
}

var MockErrorAuth = map[string]interface{}{
	"error": map[string]interface{}{
		"message": "Invalid API key",
		"code":    "AUTHENTICATION_ERROR",
		"status":  401,
	},
}

var MockErrorServer = map[string]interface{}{
	"error": map[string]interface{}{
		"message": "Internal server error",
		"code":    "SERVER_ERROR",
		"status":  500,
	},
}

// ToJSON converts a fixture to JSON bytes
func ToJSON(data interface{}) []byte {
	bytes, _ := json.Marshal(data)
	return bytes
}

File: packages/typescript/tests/mocks/fixtures.ts
import { User, Project, Bucket, FileObject } from '../../src/types';

// User fixtures
export const mockUser: User = {
    id: 'usr_123456789',
    name: 'John Doe',
    email: 'john@example.com',
    avatar_url: 'https://example.com/avatar.jpg',
    created_at: '2024-01-01T00:00:00Z',
    updated_at: '2024-01-01T00:00:00Z',
};

export const mockUsers: User[] = [
    mockUser,
    {
        id: 'usr_987654321',
        name: 'Jane Smith',
        email: 'jane@example.com',
        avatar_url: undefined,
        created_at: '2024-01-02T00:00:00Z',
        updated_at: '2024-01-02T00:00:00Z',
    },
];

export const mockUserCreate = {
    name: 'New User',
    email: 'newuser@example.com',
};

// Project fixtures
export const mockProject: Project = {
    id: 'prj_123456789',
    name: 'Test Project',
    description: 'A test project for unit tests',
    owner_id: 'usr_123456789',
    created_at: '2024-01-01T00:00:00Z',
    updated_at: '2024-01-01T00:00:00Z',
};

export const mockProjects: Project[] = [
    mockProject,
    {
        id: 'prj_987654321',
        name: 'Another Project',
        description: 'Another test project',
        owner_id: 'usr_987654321',
        created_at: '2024-01-02T00:00:00Z',
        updated_at: '2024-01-02T00:00:00Z',
    },
];

export const mockProjectCreate = {
    name: 'New Project',
    description: 'A new project',
};

// Storage fixtures
export const mockBucket: Bucket = {
    id: 'bkt_123456789',
    name: 'test-bucket',
    region: 'us-east-1',
};

export const mockBuckets: Bucket[] = [
    mockBucket,
    {
        id: 'bkt_987654321',
        name: 'public-bucket',
        region: 'eu-west-1',
    },
];

export const mockFile: FileObject = {
    id: 'file_123456789',
    bucket_id: 'bkt_123456789',
    key: 'test-file.pdf',
    size: 1024,
    content_type: 'application/pdf',
    url: 'https://storage.eclipse.dev/bkt_123456789/test-file.pdf',
};

// API response fixtures
export const mockListResponse = <T>(data: T[], hasMore = false, nextCursor?: string) => ({
    data,
    has_more: hasMore,
    next_cursor: nextCursor,
});

// Error fixtures
export const mockErrorResponse = {
    error: {
        message: 'Resource not found',
        code: 'NOT_FOUND',
        status: 404,
    },
};

export const mockValidationErrorResponse = {
    error: {
        message: 'Validation failed',
        code: 'VALIDATION_ERROR',
        status: 400,
        details: [
            { field: 'email', message: 'Invalid email format' },
        ],
    },
};

export const mockRateLimitErrorResponse = {
    error: {
        message: 'Rate limit exceeded',
        code: 'RATE_LIMIT_EXCEEDED',
        status: 429,
        retryAfter: 60,
    },
};

export const mockAuthErrorResponse = {
    error: {
        message: 'Invalid API key',
        code: 'AUTHENTICATION_ERROR',
        status: 401,
    },
};
